cmake/machinecfg/RZAnsel.cmake

```diff
@@ -0,0 +1,214 @@
+        ABSOLUTE)
+    set(CMAKE_CXX_COMPILER ${NVCC_WRAPPER} CACHE STRING "nvcc_wrapper")
+endif()
+
+string(TOLOWER "${RZANSEL_COMPILER}" RZANSEL_COMPILER_LOWER)
+set(RZANSEL_COMPILER_PREFIX ${RZANSEL_ARCH_PREFIX}/${RZANSEL_COMPILER_LOWER})
+
+if (NOT EXISTS ${RZANSEL_COMPILER_PREFIX})
+    message(WARNING "No dependencies detected for \
+        RZANSEL_COMPILER=\"${RZANSEL_COMPILER}\" at ${RZANSEL_COMPILER_PREFIX}")
+    return()
+endif()
+
+# Let the user specify the compiler if they really want to. Otherwise, point
+# to the compilers specified by the RZANSEL_ options
+if (RZANSEL_COMPILER MATCHES "GCC")
+    set(GCC_VERSION ${RZANSEL_${RZANSEL_COMPILER}_VERSION})
+    set(GCC_PREFIX /usr/tce/packages/gcc/gcc-${GCC_VERSION})
+
+    if (GCC_VERSION)
+        set(CMAKE_C_COMPILER ${GCC_PREFIX}/bin/gcc
+            CACHE STRING "gcc ${GCC_VERSION}")
+
+        set(RZANSEL_CXX_COMPILER ${GCC_PREFIX}/bin/g++)
+        if (RZANSEL_CUDA)
+            set(CMAKE_CXX_FLAGS "-ccbin ${RZANSEL_CXX_COMPILER}")
+        else()
+            set(CMAKE_CXX_COMPILER ${RZANSEL_CXX_COMPILER}
+            CACHE STRING "gcc ${GCC_VERSION}")
+        endif()
+
+        set(CMAKE_BUILD_RPATH ${GCC_PREFIX}/lib64
+            CACHE STRING "rpath libs")
+    endif()
+endif()
+
+if (NOT DEFINED CMAKE_CXX_COMPILER)
+    message(
+        FATAL_ERROR
+        "Found view on RZAnsel for compiler version ${RZANSEL_COMPILER}, but \
+        don't know how to map it to a specific compiler. Either update \
+        your Parthenon checkout or explicitly set CMAKE_C_COMPILER and \
+        CMAKE_CXX_COMPILER")
+endif()
+
+# MPI - We use the system modules since replicating them in spack can be
+# difficult.
+if (RZANSEL_COMPILER MATCHES "GCC")
+    set(MPI_ROOT /usr/tce/packages/spectrum-mpi/spectrum-mpi-${RZANSEL_MPI_DATE}-gcc-${RZANSEL_${RZANSEL_COMPILER}_VERSION}
+        CACHE STRING "MPI Location")
+endif()
+
+# clang-format
+set(CLANG_FORMAT
+    ${RZANSEL_PROJECT_PREFIX}/tools/${RZANSEL_ARCH}/bin/clang-format-8
+    CACHE STRING "clang-format-8")
+
+# Kokkos settings
+set(Kokkos_ARCH_POWER9 ON CACHE BOOL "Target Power9")
+
+if (RZANSEL_CUDA)
+    set(Kokkos_ENABLE_CUDA ON CACHE BOOL "Cuda")
+    set(Kokkos_ARCH_VOLTA70 ON CACHE BOOL "Target V100s")
+    set(NUM_GPU_DEVICES_PER_NODE ${NUM_RANKS} CACHE STRING "Number of gpu devices to use when testing if built with Kokkos_ENABLE_CUDA")
+endif()
+
+if (${MACHINE_VARIANT} MATCHES "mpi")
+  set(RZANSEL_VARIANT_PREFIX "")
+  set(NUM_MPI_PROC_TESTING ${NUM_RANKS} CACHE STRING "CI runs tests with 2 MPI ranks by default.")
+  set(MPIEXEC_EXECUTABLE "/usr/tcetmp/bin/jsrun" CACHE STRING "Command to launch MPI applications")
+  set(TEST_NUMPROC_FLAG "-c" CACHE STRING "Flag to set number of processes")
+  if (${MACHINE_VARIANT} MATCHES "cuda")
+    string(APPEND TEST_MPIOPTS "-a 1 -n 1 -g ${NUM_GPU_DEVICES_PER_NODE} -r 1 -d packed --smpiargs='-gpu'")
+  else()
+    string(APPEND TEST_MPIOPTS "-a 1 -n 1 -r 1 -d packed")
```

Comment:
Would you mind double checking this (that -c controls the number of processes) is true?
On Summit with `jsrun`  the parameter have different meaning. `-c` controls the number of cores allocated per resource set. The total number of tasks (i.e., MPI ranks) is controlled by the combination of `-n` (number or resource sets) and `-a` (number of tasks per resource set).

Reply:
Thanks for the feedback, I'll be sure to check, I probably did something wrong. I was getting errors when trying to run the tests when using -a 4, but it might just be that some of the tests are not set up to run on 4 ranks.